src/components/BudgetingList/BudgetingList.tsx

```diff
@@ -33,55 +98,98 @@ const BudgetingList = (): JSX.Element => {
     let data = [...fields];
     data.splice(index, 1);
     setFields(data);
+    budgetingData.splice(index + 6, 1);
+    setBudgetingData(budgetingData);
+    setValue("");
   };
 
   return (
     <div className="budgeting-list-component">
       <ul className="budgeting-list">
         <li>
```

Comment:
I prefer this to be generated with `Array.map`. Try using a list of objects with the key being the label and the fieldType being the type and name the variable definiting these fileds `initialData` or something